File: chazm-model/src/main/java/io/github/runtimemodels/chazm/entity/PolicyFactory.java
package io.github.runtimemodels.chazm.entity;

import runtimemodels.chazm.api.entity.Policy;
import runtimemodels.chazm.api.id.UniqueId;

/**
 * The {@linkplain PolicyFactory} interface defines the APIs for constructing {@linkplain Policy}s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
public interface PolicyFactory {

    /**
     * Constructs a {@linkplain Policy}.
     *
     * @param id the {@linkplain UniqueId} that represents the {@linkplain Policy}.
     * @return a {@linkplain Policy}.
     */
    Policy buildPolicy(UniqueId<Policy> id);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/entity/PolicyFactory.kt
package runtimemodels.chazm.model.entity

import runtimemodels.chazm.api.entity.Policy
import runtimemodels.chazm.api.entity.PolicyId
import runtimemodels.chazm.api.id.UniqueId

/**
 * The [PolicyFactory] interface defines the APIs for constructing [Policy]s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
interface PolicyFactory {
    /**
     * Constructs a [Policy].
     *
     * @param id the [UniqueId] that represents the [Policy].
     * @return a [Policy].
     */
    fun build(id: PolicyId): Policy
}
